lnbits/tasks.py

```diff
@@ -149,37 +148,36 @@ async def check_pending_payments():
     incoming = True
 
     while True:
-        async with db.connect() as conn:
-            logger.info(
-                f"Task: checking all pending payments (incoming={incoming},"
-                f" outgoing={outgoing}) of last 15 days"
-            )
-            start_time = time.time()
-            pending_payments = await get_payments(
-                since=(int(time.time()) - 60 * 60 * 24 * 15),  # 15 days ago
-                complete=False,
-                pending=True,
-                outgoing=outgoing,
-                incoming=incoming,
-                exclude_uncheckable=True,
-                conn=conn,
-            )
-            for payment in pending_payments:
-                await payment.check_status(conn=conn)
+        logger.info(
+            f"Task: checking all pending payments (incoming={incoming},"
+            f" outgoing={outgoing}) of last 15 days"
+        )
+        start_time = time.time()
+        pending_payments = await get_payments(
+            since=(int(time.time()) - 60 * 60 * 24 * 15),  # 15 days ago
+            complete=False,
+            pending=True,
+            outgoing=outgoing,
+            incoming=incoming,
+            exclude_uncheckable=True,
+        )
+        for payment in pending_payments:
+            await payment.check_status()
+            await asyncio.sleep(0.1)  # to avoid complete blocking
 
+        logger.info(
+            f"Task: pending check finished for {len(pending_payments)} payments"
+            f" (took {time.time() - start_time:0.3f} s)"
+        )
+        # we delete expired invoices once upon the first pending check
+        if incoming:
```

Comment:
- the `incoming` logic should either be outside this loop (before the while), or
- `check_pending_payments` should be split in two different functions:
   - `check_pending_outgoing_payments`
   - `check_pending_incoming_payments`

Reply:
To be safe, I won't be changing the logic of this loop in this PR, maybe some other time!